DataStructures.Tests/SegmentTrees/SegmentTreeApplyTests.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using DataStructures.SegmentTrees;
+using NUnit.Framework;
+
+namespace DataStructures.Tests.SegmentTrees
+{
+    [TestFixture]
+    public class SegmentTreeApplyTests
+    {
+        private SegmentTreeApply testTree = new SegmentTreeApply(new int[] {8, 9, 1, 4, 8, 7, 2});
+
+        [Test]
+        public void Apply_Query_Update_Query_Test()
+        {
+            Assert.AreEqual(22, testTree.Query(1, 4));
+            testTree.Apply(0, 3, 2);
+            Assert.AreEqual(new int[] {1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}, testTree.Operation);
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6278042)